cmd/healthbot/counterprobe/metrics.go

```diff
@@ -14,40 +15,41 @@ func (cp *CounterProbe) initMetrics(chainName string) error {
 	meter := global.MeterProvider().Meter("tableland")
 	cp.mBaseLabels = append([]attribute.KeyValue{attribute.String("chain_name", chainName)}, metrics.BaseAttrs...)
 
-	latencyHistogram, err := meter.SyncInt64().Histogram(metricPrefix + ".latency")
+	latencyHistogram, err := meter.Int64Histogram(metricPrefix + ".latency")
 	if err != nil {
 		return fmt.Errorf("registering latency histogram: %s", err)
 	}
 	cp.mLatencyHist = latencyHistogram
 
-	mLastCheck, err := meter.AsyncInt64().Gauge(metricPrefix + ".last_check")
+	mLastCheck, err := meter.Int64ObservableGauge(metricPrefix + ".last_check")
 	if err != nil {
 		return fmt.Errorf("registering last check gauge: %s", err)
 	}
 
-	mLastSuccessfulCheck, err := meter.AsyncInt64().Gauge(metricPrefix + ".last_successful_check")
+	mLastSuccessfulCheck, err := meter.Int64ObservableGauge(metricPrefix + ".last_successful_check")
 	if err != nil {
 		return fmt.Errorf("registering last full check gauge: %s", err)
 	}
 
-	mCounterValue, err := meter.AsyncInt64().Gauge(metricPrefix + ".counter_value")
+	mCounterValue, err := meter.Int64ObservableGauge(metricPrefix + ".counter_value")
 	if err != nil {
 		return fmt.Errorf("registering counter value gauge: %s", err)
 	}
 
 	instruments := []instrument.Asynchronous{mLastCheck, mLastSuccessfulCheck, mCounterValue}
-	if err := meter.RegisterCallback(instruments, func(ctx context.Context) {
+	if _, err := meter.RegisterCallback(func(ctx context.Context, o metric.Observer) error {
```

Comment:
This is another change that happened. The signature of the `RegisterCallback`. You'll see a lot of this change.

note: we don't need the return value. That value can be used to Unregister a callback, which is something we don't do.

Reply:
makes sense.

Reply:
why is mutex required in every callback?

Reply:
Good question! TBH I think all of them should have, but probably the ones that have are the ones we had problems with data race.